examples/set.js

```diff
@@ -1,39 +1,28 @@
-// An example explaining how to add values to a set and how to retrive them
+// An example explaining how to add values to a set and how to retrieve them
 
 import { createClient } from 'redis';
 
 const client = createClient();
 await client.connect();
 
-const setName = "user1:favorites";
+const SET_NAME = 'user1:favorites';
 
-// if you try to add any data of type other than string you will get an error saying `Invalid argument type`
-// so before adding make sure the value is of type string or convert it using toString() method
+// add an item to the set
+// set members can only be `string | Buffer` so make sure to convert values if needed,
 // https://redis.io/commands/sadd/
-await client.SADD(setName, "1");
+await client.SADD(SET_NAME, '1');
 
-// retrieve values of the set defined
+// retrieve all the members of the set in one batch (be careful using it with "large" sets)
 // https://redis.io/commands/smembers/
-const favorites = await client.SMEMBERS(setName);
+const favorites = await client.SMEMBERS(SET_NAME);
 for (const favorite of favorites) {
   console.log(favorite);
 }
 
-// alternate way to retrieve data from set
+// retrieve all the members of the set in batches using SSCAN (useful with large sets)
+// https://redis.io/commands/sscan/
 for await (const member of client.sScanIterator(setName)) {
   console.log(member);
 }
 
-// another alternate way to retrieve values from the set
-// https://redis.io/commands/sscan/
-let iCursor = 0;
-do {
-  const { cursor, members } = await client.SSCAN(setName, iCursor);
-  members.forEach((member) => {
-    console.log(member);
-  });
-  iCursor = cursor;
-} while (iCursor !== 0)
```

Comment:
I think we can keep this. Let's say for example, if user needs some kind of pagination or something then he can make use of this method.
Rest everything looks fine.

Reply:
@avinashkrishna613 this is exactly how "sScanIterator" works, there is no reason to implement it again..

Reply:
@leibale Then it's fine.